File: examples/readyaml/readyaml.go
package main

import (
	"fmt"
	. "github.com/lunfardo314/goq/readyaml"
	"os"
	"time"
)

const fname = "./examples/readyaml/QuplaTests.yml"

func main() {
	fmt.Printf("Reading from file '%v'\n", fname)

	start := time.Now()
	module, err := NewQuplaModuleFromYAML(fname)
	if err != nil {
		fmt.Printf("error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("Module '%v' has been loaded succesfully in %v\n", module.Name, time.Since(start))
	fmt.Printf("    Global types: %v\n", len(module.Types))
	fmt.Printf("    LUTs: %v\n", len(module.Luts))
	fmt.Printf("    Functions: %v\n", len(module.Functions))
	fmt.Printf("    Executable statements: %v\n", len(module.Execs))
}

File: examples/readyaml/readyaml.py
# example how to parse YAML file

import yaml
from datetime import datetime, timezone

fname = "QuplaTests.yml"

def main():
    print("reading file " + fname)
    fin = open(fname, "r")
    start = unixnow()
    yaml_string = fin.read()
    module_yaml_ir = yaml.load(yaml_string)

    print("module '{}' loaded successfully in {} seconds".format(module_yaml_ir.get("module"), unixnow()-start))
    print("  global type definitions: {}".format(len(module_yaml_ir.get("types"))))
    print("  LUTs: {}".format(len(module_yaml_ir.get("luts"))))
    print("  functions: {}".format(len(module_yaml_ir.get("functions"))))
    print("  executable statements: {}".format(len(module_yaml_ir.get("execs"))))

def unixnow():
    return int(datetime.now(tz=timezone.utc).timestamp())

if __name__ == '__main__':
    main()